### DISCUSSION TITLE
第3节课作业(2班)

### DISCUSSION THREAD

**Discussion by MING-ZCH** [InternLM/Tutorial#182] [作业]
### 基础作业

* 复现课程知识库助手搭建过程 (截图)

### 进阶作业（可选做）

* 选择一个垂直领域，收集该领域的专业资料构建专业知识库，并搭建专业问答助手，并在 [OpenXLab](https://openxlab.org.cn/apps) 上成功部署（截图，并提供应用地址）

**Comment by YanxingLiu** [comment]
## 基础作业
搭建InternLM知识库助手
<img width="1696" alt="6dddccd0a764806825ea285d03b5aa9" src="https://github.com/InternLM/tutorial/assets/42299757/6c8672d9-f9d8-4b27-8333-55d28e66d5cf">
<img width="1464" alt="e4903892282d3ca73a8d3259b09e462" src="https://github.com/InternLM/tutorial/assets/42299757/09a98c5b-0333-4a7a-b821-f29629bc37e3">
## 进阶作业
基于openmmlab开源的cv库的知识库助手
数据
![67862adce80509116528c7e02d27c48](https://github.com/InternLM/tutorial/assets/42299757/d552ca3c-6de8-43a3-a044-2590f4b60a93)
实验结果
<img width="1464" alt="e4903892282d3ca73a8d3259b09e462" src="https://github.com/InternLM/tutorial/assets/42299757/c3d88f72-538b-4dda-b6e8-92643f1d5ccc">
![8247b8fd7c045a9bae10598966870e6](https://github.com/InternLM/tutorial/assets/42299757/d36fce27-697a-4f65-932f-40e8296e5195)
![b49cc45677a8e1e3280269117bb2ad3](https://github.com/InternLM/tutorial/assets/42299757/1f4dbcc9-76b6-4f6e-9bc5-a5df1579c245)

**Comment by songyuc** [comment]
### 基础作业
-   复现课程知识库助手搭建过程

![image](https://github.com/InternLM/tutorial/assets/27288110/e56d2d51-2332-405a-b9e8-12b261a651a9)

**Comment by jinxianwei** [comment]
[基础作业](https://blog.csdn.net/jinxianwei1999/article/details/135483002?csdn_share_tail=%7B%22type%22%3A%22blog%22%2C%22rType%22%3A%22article%22%2C%22rId%22%3A%22135483002%22%2C%22source%22%3A%22jinxianwei1999%22%7D)

**Comment by pranerd** [comment]
基础作业
![interlm_langchain_1](https://github.com/InternLM/tutorial/assets/19358928/fcab26b0-51d5-458a-83c3-c6c7a5ed3265)
![interlm_langchain_2](https://github.com/InternLM/tutorial/assets/19358928/d4c0cf02-d66f-48be-b29b-495873b640f7)
进阶作用
构建一个python的知识库
![interlm_langchain_python_1](https://github.com/InternLM/tutorial/assets/19358928/f91c91a1-39bc-4114-aee3-ba9d18481fe0)
![interlm_langchain_python_2](https://github.com/InternLM/tutorial/assets/19358928/30c75c34-e23b-414a-af5d-b64327759903)

**Comment by zyy0831** [comment]
基础作业
https://blog.csdn.net/sinat_41838539/article/details/135491826?spm=1001.2014.3001.5502

**Comment by xulinui** [comment]
https://h90ag9106t.feishu.cn/docx/OeyFdw1ZYosnC3xVia5csGkpnsg

**Comment by DarthThomas** [comment]
### 作业
使用python的doc中faq的部分作为数据库，提问相关问题
<img width="1440" alt="截屏2024-01-12 21 55 54" src="https://github.com/InternLM/tutorial/assets/20266717/81b175d2-474f-4e1e-82a3-5d0b5f52f46c">
![截屏2024-01-12 22 03 11](https://github.com/InternLM/tutorial/assets/20266717/aee3ff55-4628-4362-862c-81d78beb5a39)
![截屏2024-01-12 22 07 41](https://github.com/InternLM/tutorial/assets/20266717/69deb9ac-ad56-4f54-b008-538d9c4dbc1d)

**Comment by ljqiff** [comment]
基础作业
![image](https://github.com/InternLM/tutorial/assets/29425919/f9e41f2c-01b9-411b-8168-40671d8b8a79)

进阶作业（可选做）
![image](https://github.com/InternLM/tutorial/assets/29425919/434eac90-888d-4e3b-b9b7-3e6826bf21b9)
https://openxlab.org.cn/apps/detail/ljqiff/CarLLMTest

**Comment by qihong2019** [comment]
基础作业
![image](https://github.com/InternLM/tutorial/assets/47970679/bf5cd844-9115-448c-ad5e-6b2b6e92e238)
![image](https://github.com/InternLM/tutorial/assets/47970679/d9505d84-96a6-4acc-aa38-759850dca9cc)

进阶作业
![image](https://github.com/InternLM/tutorial/assets/47970679/667e7d5a-50ad-48e7-aed5-0ab7a75f6d3e)
![image](https://github.com/InternLM/tutorial/assets/47970679/10a117b7-6fba-48e8-a329-3a0935a0bf80)

**Comment by kyrie2to11** [comment]
[InterLM L-3: 基于 InternLM 和 LangChain 搭建知识库](https://zhuanlan.zhihu.com/p/676897517) 基础作业打卡~

**Comment by kishiokon** [comment]
基础作业:完成RAG+LLM demo
![US8$R90XL1YMBS53MJKE8QD](https://github.com/InternLM/tutorial/assets/92700495/53a7a45c-2ba4-4786-9761-e5b119401e0f)
进阶作业:挂载Optimization数据库,测试demo
![HBSAQG3@KD9@II0$OVZU(0Y](https://github.com/InternLM/tutorial/assets/92700495/5b895322-4c04-4019-882a-554a7289d3c2)
![RGI3) 0K0`$XE@SEUIIP(R](https://github.com/InternLM/tutorial/assets/92700495/e60366f1-4982-4557-a41b-04c20782b9f3)

**Comment by dcuplover** [comment]
[基础作业](https://www.yuque.com/u174176/new86d/kq2abfnaqg8dty6p)

**Comment by CoderJackZhu** [comment]
基础作业
复现课程知识库助手搭建过程 (截图)
![image](https://github.com/InternLM/tutorial/assets/42195616/ba09079c-d614-4511-89d5-d01cf7a3d461)

进阶作业（可选做）
选择一个垂直领域，收集该领域的专业资料构建专业知识库，并搭建专业问答助手，并在 [OpenXLab](https://openxlab.org.cn/apps) 上成功部署（截图，并提供应用地址）
这里选用的是仓库[amusi/CVPR2023-Papers-with-Code: CVPR 2023 论文和开源项目合集 (github.com)](https://github.com/amusi/CVPR2023-Papers-with-Code)的内容作为语料库
![image](https://github.com/InternLM/tutorial/assets/42195616/67d846ea-17cb-4df7-856d-f2ae8e547ca0)

**Comment by ZixinxinWang** [comment]
1.基础作业
![image](https://github.com/InternLM/tutorial/assets/152623177/32dd852e-6705-4984-a271-86f687cfbfd3)

2.进阶作业
<img width="924" alt="image" src="https://github.com/InternLM/tutorial/assets/152623177/4e25b632-d63e-410b-b616-ab645e457142">

**Comment by phdbrianlee** [comment]
基础作业：

**Reply by MING-ZCH** [reply]
同学请提供作业链接/截图哦~

**Comment by yanxiaoqi932** [comment]
![20240112214817](https://github.com/InternLM/tutorial/assets/55906038/d1781753-fe94-4e76-88d8-7ea5c6589864)
![20240112214828](https://github.com/InternLM/tutorial/assets/55906038/df8e3f7a-7e41-4bc6-9351-2185efb4206c)
![20240112220800](https://github.com/InternLM/tutorial/assets/55906038/34057868-106f-4fe2-99c5-046ade9d28b0)
![20240115132506](https://github.com/InternLM/tutorial/assets/55906038/d2753cfb-1101-4ba1-84f9-01dc3a0e367c)

**Comment by kaoyansoft123** [comment]
基础作业：
![image](https://github.com/InternLM/tutorial/assets/152302734/728115a5-8682-4733-a06e-2ccb1ae3a7fc)
![image](https://github.com/InternLM/tutorial/assets/152302734/36132d32-5328-4105-a335-19cc4174dc0f)

**Comment by WanpengXu** [comment]
[[作业] 书生·浦语大模型实战营-第三节](https://b07ofnm9xj7.feishu.cn/wiki/ZtQXwB4NFirXjwk47ebc6BG5nMd)

**Comment by Mr-Vincenyang** [comment]
[作业](https://zhuanlan.zhihu.com/p/677682069)

**Comment by zyy0831** [comment]
第三课作业
https://blog.csdn.net/sinat_41838539/article/details/135491826?spm=1001.2014.3001.5501

**Comment by KylinCarol** [comment]
![image](https://github.com/InternLM/tutorial/assets/135086891/1ea92822-cfbd-4b47-bfec-5c7710abbc7e)

**Comment by PrinceRay7** [comment]
第三节课作业
https://github.com/PrinceRay7/InternLM_HW/blob/main/hw3/homework/homework.md
